src/services/safes.rs

```diff
@@ -1,11 +1,116 @@
+        safe_config: safe_info_ex,
+        safe_state: SafeLastChanges {
+            collectibles: get_last_collectible(context, safe_address).unwrap_or(0),
+            tx_queued: get_last_queued_tx(context, safe_address).unwrap_or(0),
+            tx_history: get_last_history_tx(context, safe_address).unwrap_or(0),
+        },
+    };
+
+    Ok(safe_state)
+}
+
+fn get_last_collectible(context: &Context, safe_address: &String) -> ApiResult<i64> {
+    let url = format!(
+        "{}/v1/safes/{}/transfers/?\
+        &ordering=executionDate\
+        &limit=1",
+        base_transaction_service_url(),
+        safe_address,
+    );
+
+    let body = RequestCached::new(url)
+        .request_timeout(transaction_request_timeout())
+        .execute(context.client(), context.cache())?;
+    let transaction: Page<Transfer> = serde_json::from_str(&body)?;
+
+    log::debug!("{:#?}", &transaction);
+    transaction
+        .results
+        .get(0)
+        .as_ref()
+        .map(|transfer| match transfer {
+            Transfer::Erc721(transfer) => transfer.execution_date.timestamp(),
+            Transfer::Erc20(transfer) => transfer.execution_date.timestamp(),
+            Transfer::Ether(transfer) => transfer.execution_date.timestamp(),
+            Transfer::Unknown => 0,
+        })
+        .ok_or(api_error!("Couldn't get tx timestamps"))
+}
+
+fn get_last_queued_tx(context: &Context, safe_address: &String) -> ApiResult<i64> {
+    let url = format!(
+        "{}/v1/safes/{}/multisig-transactions/?\
+        &ordering=nonce,submissionDate\
+        &executed=false\
+        &trusted=true\
+        &limit=1",
+        base_transaction_service_url(),
+        safe_address,
+    );
+
+    let body = RequestCached::new(url)
+        .request_timeout(transaction_request_timeout())
+        .execute(context.client(), context.cache())?;
+    let transaction: Page<MultisigTransaction> = serde_json::from_str(&body)?;
+
+    log::debug!("{:#?}", &transaction);
+    transaction
+        .results
+        .get(0)
+        .as_ref()
+        .map(|tx| tx.submission_date.timestamp())
+        .ok_or(api_error!("Couldn't get tx timestamps"))
+}
+
+fn get_last_history_tx(context: &Context, safe_address: &String) -> ApiResult<i64> {
+    let mut info_provider = DefaultInfoProvider::new(context);
+
+    let url = format!(
+        "{}/v1/safes/{}/all-transactions/?\
+        &ordering=executionDate
+        &queued=false\
+        &executed=true",
+        base_transaction_service_url(),
+        safe_address
+    );
+
+    let body = RequestCached::new(url)
+        .request_timeout(transaction_request_timeout())
+        .execute(context.client(), context.cache())?;
+    let transaction: Page<Transaction> = serde_json::from_str(&body)?;
+
+    log::debug!("{:#?}", &transaction);
+    transaction
+        .results
+        .get(0)
+        .as_ref()
+        .and_then(|tx| {
+            tx.to_transaction_summary(&mut info_provider, safe_address)
```

Comment:
If possible I would like to avoid this as this might pull all kind of additional information (and therefore trigger requests) that we don't need here

Reply:
Here if we use the backend tx the date selection tree changes a little. You will see in the next commit.

Reply:
We can either default to `0` or `submission_date`

Reply:
Yeah we have to do a little bit more mapping, but I think it is better. I would go with the solution from the commit you mentioned